pkg/util/exclude_kubernetes.go

```diff
@@ -0,0 +1,106 @@
+func ExcludeKubernetesResources(fs afero.Afero, basePath string, excludedResources []string) error {
+	for _, excludedResource := range excludedResources {
+		err := ExcludeKubernetesResource(fs, basePath, excludedResource)
+		if err != nil {
+			return errors.Wrapf(err, "excluding %s from %s", excludedResource, basePath)
+		}
+	}
+	return nil
+}
+
+// exclude the provided kubernetes resource file from the kustomization.yaml at basePath, or from bases imported from that.
+func ExcludeKubernetesResource(fs afero.Afero, basePath string, excludedResource string) error {
+	kustomizeYaml, err := fs.ReadFile(filepath.Join(basePath, "kustomization.yaml"))
+	if err != nil {
+		return errors.Wrapf(err, "read kustomization yaml in %s", basePath)
+	}
+
+	kustomization := types.Kustomization{}
+	err = yaml.Unmarshal(kustomizeYaml, &kustomization)
+	if err != nil {
+		return errors.Wrapf(err, "unmarshal kustomization yaml from %s", basePath)
+	}
+
+	excludedResource = strings.TrimPrefix(excludedResource, string(filepath.Separator))
+
+	newResources := []string{}
+	for _, existingResource := range kustomization.Resources {
+		if existingResource == excludedResource {
+			fmt.Printf("excluding %s from %s", excludedResource, basePath)
+		} else {
+			newResources = append(newResources, existingResource)
+		}
+	}
+
+	if len(newResources) != len(kustomization.Resources) {
+		kustomization.Resources = newResources
+
+		// write updated kustomization to disk - resource has been removed
+		kustomizeYaml, err = yaml.Marshal(kustomization)
+		if err != nil {
+			return errors.Wrapf(err, "marshal kustomization yaml from %s", basePath)
+		}
+
+		err = fs.WriteFile(filepath.Join(basePath, "kustomization.yaml"), kustomizeYaml, 0644)
+		if err != nil {
+			return errors.Wrapf(err, "write kustomization yaml to %s", basePath)
+		}
+		return nil
+	} else {
+		// check if the resource is already removed from this dir
+		alreadyRemoved := false
+		err = fs.Walk(basePath, func(path string, info os.FileInfo, err error) error {
+			if !info.IsDir() {
+				relPath, err := filepath.Rel(basePath, path)
+				if err != nil {
+					return errors.Wrapf(err, "get relative path to %s from %s", path, basePath)
+				}
+				fmt.Printf("path %s, excluded %s\n", relPath, excludedResource)
+				if relPath == excludedResource {
+					alreadyRemoved = true
+				}
+			}
+			return nil
+		})
+		if err != nil {
+			return errors.Wrapf(err, "walk files in %s", basePath)
+		}
+		if alreadyRemoved {
+			// the file to be removed exists within this base dir, and not within the kustomization yaml
+			return nil
+		}
+
+		for _, newBase := range kustomization.Bases {
+			newBase = filepath.Clean(filepath.Join(basePath, newBase))
+			cleanBase := strings.ReplaceAll(newBase, string(filepath.Separator), "-")
+
+			fmt.Printf("newBase: %s cleanBase: %s\n", newBase, cleanBase)
+
+			if strings.HasPrefix(excludedResource, cleanBase) {
```

Comment:
Substring matching like this on files only works if `cleanBase` ends with a file separator because it has to match a full directory name, right?   Does it always end with a file separator?

Reply:
Other way around - this is testing if 'excludedResource' has a prefix of 'cleanBase'

Reply:
right, the other way.... still the same question though

Reply:
Neither `cleanBase` or `excludedResource` end with a seperator. `cleanBase` (in practice) will be things like `another-base` for a base within `another/base`.